lib/bundler/ruby_version.rb

```diff
@@ -100,6 +100,15 @@ def self.system
       @ruby_version ||= RubyVersion.new(RUBY_VERSION.dup, RUBY_PATCHLEVEL.to_s, ruby_engine, ruby_engine_version)
     end
 
+    def to_gem_version_with_patchlevel
+      @gem_version_with_patch ||= begin
+        patch_number = @patchlevel ? Gem::Requirement.create(@patchlevel).requirements.first.last : 0
```

Comment:
I think it's exactly the same. I'll change it, yours is simpler.
The purpose is to avoid raising and catching an exception when `@patchlevel` is nil. Although the exception is still risen and caught if `@patchlevel` is negative or some other invalid value.

I think that performance-wise it better to handle these kind of situations using conditions instead of catching the exception. But if I were to fully validate the `@patchlevel`, it would be like duplicating the code in Gem::Version that already does that.

I think that the code that just catches the exception instead of using conditions would be easier to understand:

```
@gem_version_with_patch ||= begin
  Gem::Version.create("#{@gem_version}.#{@patchlevel}")
rescue BadRequirementError
  Gem::Version.create(@gem_version)
end
```

Which one would you consider to be better?